### PR TITLE
NORTHWEST_4 - SALVADOR CUEVAS GARCIA - JAVASCRIPT CORE 1 COURSEWORK WEEK 2

### PR DESCRIPTION
**Pull request by Salvacg737**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

**Review by erin-switchstitch — COMMENTED** [review]
exercises/B-boolean-literals/exercise.js : 

Perhaps look at using the codeYourFutureIsGreat variable to set the boolean value for the other variables

**Review by erin-switchstitch — COMMENTED** [review]
exercises/C-comparison-operators/exercise.js :

Please check line 13 - it works if the maxCapacity is 25 and the number of students and mentors is 25, but will not work if the number of students and mentors is e.g. 24

**Review by erin-switchstitch — COMMENTED** [review]
exercises/D-logical-operators/exercise.js :

Because of the way you have written this the >5 is only checking against cssLevelAbove5. Instead you should do something like this : (htmlLevelAbove5 > 5) && (cssLevelAbove5 > 5).

However for this exercise you don't need  " >5 " at the end of  " htmlLevelAbove5 && cssLevelAbove5 > 5; " . You have already checked if they are above 5 in the previous variables. Also if you console.log(htmlLevelAbove5) then you'll see that it is a boolean value (true or false), so you wouldn't want to try and compare it with a number e.g. true > 5

**Review by erin-switchstitch — COMMENTED** [review]
I've checked through all the exercises ( B to J ) and left some notes on some slight changes you may want to make. Apart from those changes, everything else looks correct. Well done !!!!

File: exercises/D-logical-operators/exercise2.js [hunk 1]
```diff
@@ -5,8 +5,33 @@
...
+  return false;
+}
+
+function isShortName(name) {
+  if (name.length < 7) {
+    return true;
+  }
+  return false;
+}
+
```
Comment (csfilipinyi) [inline exercises/D-logical-operators/exercise2.js:28]: plz. try to think about how you can make more concise these functions
hint:
let name = "Salva"
let test = name.length < 7
console.log(typeof test)

File: mandatory/2-function-creation.js [hunk 2]
```diff
@@ -25,7 +34,13 @@ Write a function that:
...
 - makes all strings all lowercase
 */
-function tidyUpStrings(arrayOfStrings) {}
+function tidyUpStrings(arrayOfStrings) {
+  let noWhiteSpacesString = arrayOfStrings.trim();
+  let noForwardSlashesString = noWhiteSpacesString.replace("/", ""));
+  let lowerCaseString = noForwardSlashesString.toLoweCase();
+
+  return lowerCaseString;
+}
```
Comment (csfilipinyi) [inline mandatory/2-function-creation.js:44]: plz. double check this function, and think about the type of the arrayOfStrings, and whether it has trim function

File: exercises/D-logical-operators/exercise.js [hunk 3]
```diff
@@ -11,14 +11,14 @@ var cssLevel = 4;
...
+var htmlLevelAbove5 = htmlLevel > 5;
+var cssLevelAbove5 = cssLevel > 5;
 
 // Finish the next two statement
 // Use the previous variables and logical operators
 // Do not "hardcode" the answers
-var cssAndHtmlAbove5;
-var cssOrHtmlAbove5;
+var cssAndHtmlAbove5 = htmlLevelAbove5 && cssLevelAbove5 > 5;
+var cssOrHtmlAbove5 = htmlLevelAbove5 || cssLevelAbove5 > 5;
```
Comment (csfilipinyi) [inline exercises/D-logical-operators/exercise.js:22]: plz. test the value and type of the cssLevelAbove5 > 5, cssLevelAbove5 is already a boolean, try to test with different value of cssLevel and you will see why it's wrong

File: exercises/F-predicates/exercise.js [hunk 4]
```diff
@@ -6,10 +6,18 @@
 */
 
 // Finish the predicate function to test if the passed number is negative (less than zero)
-function isNegative(number) {}
+function isNegative(number) {
+  if (number < 0) {
+    return true;
+  } else return false;
+}
```
Comment (csfilipinyi) [inline exercises/F-predicates/exercise.js:14]: plz. think about the type and value of the expression number < 0, and how you can simplify this function

File: mandatory/1-fix-functions.js [hunk 5]
```diff
@@ -13,17 +13,17 @@
 function mood() {
   let isHappy = true;
 
-  if (isHappy) {
+  if (isHappy == true) {
```
Comment (csfilipinyi) [inline mandatory/1-fix-functions.js:16]: this is not correct, think about how this function is called

File: mandatory/3-playing-computer.js [hunk 6]
```diff
@@ -8,11 +8,11 @@
...
-  4. How many times is "f1" called?
-  5. How many times is "f2" called?
-  6. What value does the "a" parameter take in the first "f1" call?
-  7. What is the value of the "a" outer variable when "f1" is called for the first time?
+  3. What is printed to the console? 2, 6, b, 
+  4. How many times is "f1" called? 1
+  5. How many times is "f2" called? 1
+  6. What value does the "a" parameter take in the first "f1" call? 7
+  7. What is the value of the "a" outer variable when "f1" is called for the first time? 6
 */
```
Comment (csfilipinyi) [inline mandatory/3-playing-computer.js:16]: plz. double check it again, the answers are not correct

**Review by csfilipinyi — COMMENTED** [review]
well done, see my comments inside the code

**Pull request closed without merging**